internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java

```diff
@@ -69,25 +88,41 @@ public synchronized <K, V> VeniceChangelogConsumer<K, V> getChangelogConsumer(St
             newStoreChangelogClientConfig,
             consumer != null ? consumer : new KafkaConsumer<>(newStoreChangelogClientConfig.getConsumerProperties()));
       }
+      newStoreChangelogClientConfig.getInnerClientConfig().setMetricsRepository(metricsRepository);
       return new VeniceAfterImageConsumerImpl(
           newStoreChangelogClientConfig,
           consumer != null ? consumer : new KafkaConsumer<>(newStoreChangelogClientConfig.getConsumerProperties()));
     });
   }
 
   private String getViewClass(String storeName, String viewName, D2ControllerClient d2ControllerClient, int retries) {
-    StoreResponse response =
-        d2ControllerClient.retryableRequest(retries, controllerClient -> controllerClient.getStore(storeName));
-    if (response.isError()) {
-      throw new VeniceException(
-          "Couldn't retrieve store information when building change capture client for store " + storeName);
-    }
-    ViewConfig viewConfig = response.getStore().getViewConfigs().get(viewName);
-    if (viewConfig == null) {
-      throw new VeniceException(
-          "Couldn't retrieve store view information when building change capture client for store " + storeName
-              + " viewName " + viewName);
-    }
-    return viewConfig.getViewClassName();
+    return this.defaultViewClassGetter.apply(storeName, viewName, d2ControllerClient, retries);
+  }
+
+  private ViewClassGetter getDefaultViewClassGetter() {
+    return (String storeName, String viewName, D2ControllerClient d2ControllerClient, int retries) -> {
+      StoreResponse response =
+          d2ControllerClient.retryableRequest(retries, controllerClient -> controllerClient.getStore(storeName));
+      if (response.isError()) {
+        throw new VeniceException(
+            "Couldn't retrieve store information when building change capture client for store " + storeName);
+      }
+      ViewConfig viewConfig = response.getStore().getViewConfigs().get(viewName);
+      if (viewConfig == null) {
+        throw new VeniceException(
+            "Couldn't retrieve store view information when building change capture client for store " + storeName
+                + " viewName " + viewName);
+      }
+      return viewConfig.getViewClassName();
+    };
+  }
+
+  protected void setViewClassGetter(ViewClassGetter viewClassGetter) {
```

Comment:
Nit: When we set `defaultViewClassGetter` it is not default anymore. Besides this set method is only for testing purpose?

Reply:
Yeah it's pretty much only for testing.  It was either this or do quite a lot of refactoring in our d2controllerclient.

Reply:
Got it! Maybe just use the name of `viewClassGetter`?

Reply:
done